### PR TITLE
Add validate-config command

### LINKED ISSUE #16: Add `validate-config` command
- [x] Add the new command
- [x] Add tests for the new command
- [x] Fix output of validation error messages in Config class
- [x] Add unit tests for Config class
- [x] Add unit tests for Github class
- [x] Tidy up fixtures for tests
- [x] Address TODOs in code
- [x] Fix regex used by `Github#extractOwnerAndRepo`

### PR DESCRIPTION
**Pull request by simonplend**
Resolves #16.

### CODE REVIEW

File: src/commands/validate-config.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+			describe: 'Path to JSON configuration (URL or local filepath)',
+			demandOption: true,
+			type: 'string'
+		});
+};
+
+const handler = async (argv) => {
+	try {
+		await main(argv);
+	} catch (err) {
```
Comment (sjparkinson) [inline src/commands/validate-config.js]: If we get an error on `config.load()`, will we actually get a exit code of 1, or does the handling of the catch in main not prevent that?

File: test/lib/github.test.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	const githubRepoString = `https://github.com/${expectedOwner}`;
+
+	const github = new Github();
+
+	expect(() => {
+		github.extractOwnerAndRepo(githubRepoString);
+	}).toThrowError('Github#extractOwnerAndRepo');
+});
+
+// TODO: Add more tests when fixing the regex in `extractOwnerAndRepo`
```
Comment (sjparkinson) [inline test/lib/github.test.js]: I think I have what you need.

```js
/^(?:.+github\.com(?:\/|:))?(?<owner>[\w-]+)\/(?<repo>[\w-]+)/
```

https://regex101.com/r/1CQW9Q/1

Comment (simonplend) [reply test/lib/github.test.js]: :heart_eyes:

:bowing_man:

Thank you! Please go ahead and add that in. then I'll put in a couple more tests.

File: src/commands/validate-config.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	});
+
+	try {
+		await config.load();
+
+		logger.info(`Config: Read from ${config.sourceDescription}\n`);
+		logger.success('Config is valid');
+	} catch (err) {
+		logger.error('Config is invalid');
+		logger.error(err.message);
```
Comment (simonplend) [inline src/commands/validate-config.js]: Throw these errors instead of logging them here.

File: src/commands/validate-config.js (hunk 1 above)
Comment (simonplend) [reply src/commands/validate-config.js]: Good _`catch`_.

(Sam is next to me saying "oh dear". This was totally worth it.)

**Pull request merged**